src/IssueViz.Security.UnitTests/Taint/TaintStoreTests.cs

```diff
@@ -0,0 +1,118 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Security.Taint;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.UnitTests.Taint
+{
+    [TestClass]
+    public class TaintStoreTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            var issueStoreObserver = new Mock<IIssueStoreObserver>();
+            issueStoreObserver
+                .Setup(x => x.Register(It.IsAny<ReadOnlyObservableCollection<IAnalysisIssueVisualization>>()))
+                .Returns(Mock.Of<IDisposable>());
+
+            MefTestHelpers.CheckTypeCanBeImported<TaintStore, ITaintStore>(null, new[]
+            {
+                MefTestHelpers.CreateExport<IIssueStoreObserver>(issueStoreObserver.Object)
+            });
+        }
+
+        [TestMethod]
+        public void Ctor_RegisterToIssueStoreObserver()
+        {
+            var issueStoreObserver = new Mock<IIssueStoreObserver>();
+            CreateTestSubject(issueStoreObserver.Object);
+
+            issueStoreObserver.Verify(x => x.Register(It.IsAny<ReadOnlyObservableCollection<IAnalysisIssueVisualization>>()), Times.Once);
+            issueStoreObserver.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void Dispose_InnerIssueVizStoreIsDisposed()
+        {
+            var unregisterMock = new Mock<IDisposable>();
+            var issueStoreObserver = new Mock<IIssueStoreObserver>();
+            issueStoreObserver
+                .Setup(x => x.Register(It.IsAny<ReadOnlyObservableCollection<IAnalysisIssueVisualization>>()))
+                .Returns(unregisterMock.Object);
+
+            var testSubject = CreateTestSubject(issueStoreObserver.Object);
+
+            unregisterMock.VerifyNoOtherCalls();
+
+            testSubject.Dispose();
+
+            unregisterMock.Verify(x => x.Dispose(), Times.Once);
+        }
+
+        [TestMethod]
+        public void GetAll_ReturnsReadOnlyObservableWrapper()
```

Comment:
Is it worth testing that `GetAll` returns the same instance each time? I`m don't think it matters to the caller, but it is what we now expect from the implementation.

Reply:
I don't think it matters. There's no difference in the store nor in the callers, it can be implemented either way.